content/en/blog/2024/building-trust-into-os-images-for-coco.md

```diff
@@ -0,0 +1,128 @@
+	1 : 0x33D454DFCF5E46C0B7AFD332272E72ADC3D1A86CCAE25AA98DD475C9FCA36CFC
+	2 : 0x7842C772A64365B48AC733EDEE9B131DF5F0E71EA95074F80E32450995C5773D
+	3 : 0x3D458CFE55CC03EA1F443F1562BEEC8DF51C75E14A9FCF9A7234A13F198E7969
+	4 : 0x22B156BE656EED7542AB03CC76DCC8A82F2A31044B5F17B3B8A388CB8DB37850
+	5 : 0x3F72C8A7A38564991898859F725D12E5BE64CBD26265BC8F5E39CBE1101EBD49
+	6 : 0x3D458CFE55CC03EA1F443F1562BEEC8DF51C75E14A9FCF9A7234A13F198E7969
+	7 : 0x65CAF8DD1E0EA7A6347B635D2B379C93B9A1351EDC2AFC3ECDA700E534EB3068
+	8 : 0x0000000000000000000000000000000000000000000000000000000000000000
+	9 : 0x8E74577DC5814F2EBF094988CB2E789F1D637B4D43930F3714500F9E2E65615D
+	10: 0x961D21A6CB38D377F951748BA7B8DD05A2E1BA6C712BB34EF7A39C5862721F1E
+	11: 0x9DBA7A9D3C5200B0E526112151BBD23D77006CBFCF290CFA6249601CA9812608
+```
+
+If we boot the same image on a Confidential VM in Azure's cloud, we'll see different measurements. This is expected since the early boot stack does not match our reference setup:
+
+```bash
+tpm2_pcrread sha256:0,1,2,3,4,5,6,7,8,9,10,11
+  sha256:
+	0 : 0x782B20B10F55CC46E2142CC2145D548698073E5BEB82752C8D7F9279F0D8A273
+	1 : 0x3D458CFE55CC03EA1F443F1562BEEC8DF51C75E14A9FCF9A7234A13F198E7969
+	2 : 0x3D458CFE55CC03EA1F443F1562BEEC8DF51C75E14A9FCF9A7234A13F198E7969
+	3 : 0x3D458CFE55CC03EA1F443F1562BEEC8DF51C75E14A9FCF9A7234A13F198E7969
+	4 : 0xC7BB081502F18392EB5837951A9BA48E9DB23F91DE39A9AF8B2B29C333D71EA0
+	5 : 0x0358DC1195BBDD59E3C556A452E292A6E7ECF11408BE7DAEC6776E678BEBEC23
+	6 : 0x531086506EADC75D0E540F516D68E03095E5700FE8F1BD0F840025B07A3AB4F7
+	7 : 0x64CDD65955B69C5ADD78577E32BFE52DDF9ADBF240977AEA39703908F4F6D8BA
+	8 : 0x0000000000000000000000000000000000000000000000000000000000000000
+	9 : 0x8E74577DC5814F2EBF094988CB2E789F1D637B4D43930F3714500F9E2E65615D
+	10: 0x5A7ACDE0EF2AB221551CB24CCFDB7AE959047E3C0E0C39427D329992A9C7FDDF
+	11: 0x9DBA7A9D3C5200B0E526112151BBD23D77006CBFCF290CFA6249601CA9812608
+```
+
+We can identify the common PCRs between the measurements in a cloud VM and those that we gathered in our reference setup. Those are good candidates to include them as [reference values](https://github.com/confidential-containers/trustee/tree/main/attestation-service/rvps) in a relying party against which a TEE's evidence can be verified.
+
+```bash
+grep -F -x -f pcr_reference.txt pcr_cloud.txt
+3 : 0x3D458CFE55CC03EA1F443F1562BEEC8DF51C75E14A9FCF9A7234A13F198E7969
+8 : 0x0000000000000000000000000000000000000000000000000000000000000000
+9 : 0x8E74577DC5814F2EBF094988CB2E789F1D637B4D43930F3714500F9E2E65615D
+11: 0x9DBA7A9D3C5200B0E526112151BBD23D77006CBFCF290CFA6249601CA9812608
+```
+
+## Conclusion
+
+We have learned how we can build trusted OS host images for Confidential Containers in different scenarios. The article wants to be illustrate how a chain of trust can be established using concrete examples and tools. We haven't described the scenarios technologies comprehensively, each of those would be worth their own deep-dive blog entry.
```

Comment:
@mkulke nice doc and very useful.
One doubt I got w.r.to verification of the PCR values by KBS. In KBS/AS, is it possible to configure which PCR values to verify from the ones sent in the attestation claims ?

Reply:
thx for the review! I'll refine it a bit a more. regarding your question, yes in the latest revisions of the kbs/gc stack that should be possible: 

```json
[
  {
    "name": "aztdxvtpm.tpm.pcr11",
    "expired": "2025-01-01T00:00:00Z",
    "hash-value": [
      {
        "alg": "sha256",
        "value": "9dba7a9d3c5200b0e526112151bbd23d77006cbfcf290cfa6249601ca9812608"
      }
    ]
  }
]
```